utils/buildactions/install_discord.lua

```diff
@@ -0,0 +1,250 @@
+	else
+		errormsg("discord-rpc consistency checks failed.", ("Expected %s, got %s"):format(DISCORD_HASH, downloaded_hash))
+		os.exit(1)
+		return
+	end
+
+	-- Seriously abort now if we're not using Windows
+	if os.host() ~= "windows" then
+		return
+	end
+
+	-- Delete old discord-rpc files
+	if has_discord_dir then
+		if not os.rmdir(DISCORD_PATH) then
+			errormsg("ERROR: Could not delete discord folder")
+			os.exit(1)
+			return
+		end
+	end
+
+	if not os.mkdir(DISCORD_PATH) then
+		errormsg("ERROR: Could not create discord folder (2)")
+		os.exit(1)
+		return
+	end
+
+	-- Extract zip
+	if not os.extract_archive(archive_path, DISCORD_PATH, true) then
+		errormsg("ERROR: Could not extract .zip")
+		os.exit(1)
+		return
+	end
+
+	-- Move all files from discord-rpc*/* to ./
+	os.expanddir_wildcard(DISCORD_PATH.."discord-rpc*", DISCORD_PATH)
+end
+
+local function check_rapid(upgrade)
+	local has_rapid_dir = os.isdir(RAPID_PATH)
+
+	-- Check file hash
+	local archive_path = RAPID_TEMP
+	local hash_passed = os.isfile(archive_path) and os.sha256_file(archive_path) == RAPID_HASH
+	if hash_passed then
+		print("rapidjson consistency checks succeeded")
+
+		if has_rapid_dir then
+			return
+		end
+	else
+		-- Download rapidjson
+		print("Downloading rapidjson " .. RAPID_VERSION ..  "...")
+		if not http.download_print_errors(make_download_url(RAPID_URL, RAPID_VERSION, RAPID_EXT), archive_path) then
+			os.exit(1)
+			return
+		end
+	end
+
+	local downloaded_hash = os.sha256_file(archive_path)
+	if upgrade then
+		print("New rapidjson hash is:", downloaded_hash)
+		RAPID_HASH = downloaded_hash
+
+		io.write(("Update `install_discord.lua` file? (Y/n) "):format(version))
+		local input = io.read():lower()
+		if (input == "y" or input == "yes") then
+			update_install_discord("RAPID", RAPID_VERSION, downloaded_hash)
+		end
+	end
+
+	if downloaded_hash == RAPID_HASH then
+		print("rapidjson consistency checks succeeded")
+	else
+		errormsg("rapidjson consistency checks failed.", ("Expected %s, got %s"):format(RAPID_HASH, downloaded_hash))
+		os.exit(1)
+		return
+	end
+
+	-- Seriously abort now if we're not using Windows
+	if os.host() ~= "windows" then
+		return
+	end
+
+	-- Delete old rapidjson files
+	if has_discord_dir then
+		if not os.rmdir(RAPID_PATH) then
+			errormsg("ERROR: Could not delete rapidjson folder")
+			os.exit(1)
+			return
+		end
+	end
+
+	if not os.mkdir(RAPID_PATH) then
+		errormsg("ERROR: Could not create rapidjson folder (2)")
+		os.exit(1)
+		return
+	end
+
+	-- Extract zip
+	if not os.extract_archive(archive_path, RAPID_PATH, true) then
+		errormsg("ERROR: Could not extract .zip")
+		os.exit(1)
+		return
+	end
+
+	-- Move all files from rapidjson*/* to ./
+	os.expanddir_wildcard(RAPID_PATH.."rapidjson*", RAPID_PATH)
+end
+
+newaction {
+	trigger = "install_discord",
+	description = "Downloads and installs discord-rpc",
+
+	execute = function(...)
+		local upgrade = _ARGS[1] == "upgrade"
```

Comment:
Please rename `upgrade` to `should_upgrade` or `is_upgrade_requested` for readability.